app/file.go

```diff
@@ -71,14 +72,26 @@ func ReadFile(path string) ([]byte, *model.AppError) {
 		accessKey := utils.Cfg.FileSettings.AmazonS3AccessKeyId
 		secretKey := utils.Cfg.FileSettings.AmazonS3SecretAccessKey
 		secure := *utils.Cfg.FileSettings.AmazonS3SSL
+		encrypted := *utils.Cfg.FileSettings.EnableClientSideEncryption
 		signV2 := *utils.Cfg.FileSettings.AmazonS3SignV2
 		s3Clnt, err := s3New(endpoint, accessKey, secretKey, secure, signV2)
 		if err != nil {
 			return nil, model.NewLocAppError("ReadFile", "api.file.read_file.s3.app_error", nil, err.Error())
 		}
 		bucket := utils.Cfg.FileSettings.AmazonS3Bucket
-		minioObject, err := s3Clnt.GetObject(bucket, path)
-		defer minioObject.Close()
```

Comment:
@coreyhulen  talked to the minio guys, and it looks like they have a bug.  so we're dependent on https://github.com/minio/minio-go/pull/699 which I'm told should be merged in today.  I've got the updated code ready to go as soon as they merge in their fix.